File: AI/intelligence.js
'use strict';

module.exports = class Intelligence {
    constructor(app){
        this.app = app;
    }

    start() {
        //Load map
        const Map = require('./maps/map_2021');
        this.map = new Map(this.app);
        this.map.init();
        this.app.logger.log("Map Loaded");

        //Load goals
        this.goals = require('./goals/goals').goals;
        this.app.logger.log("Goals Loaded");

        //Load Robot
        const Robot = require('./robots/robot_2022');
        this.app.logger.log("Robot Loaded");
        let robot = new Robot(this.app);

        //Running goals
        for(const goal of this.goals){
            this.app.logger.log("Running: "+goal.name)
            for(const action of goal.actions){
                let success = robot.run(action);
                this.app.logger.log(success?"Done":"Failed");
            }
        }
    }
}

File: pythonAI/intelligence.py
import json
from robots.robot_2022 import Robot2022


class Intelligence():

    def __init__(self, app):
        self.app = app

    def start(self):
        
        with open('./pythonAI/goals/goals2022.json') as file:
            goals = json.load(file)
            self.app['logger'].info("Goals loaded")

        robot = Robot2022(self.app)
        self.app['logger'].info("Robot loaded")

        for goal in goals:
            self.app['logger'].info("Running: "+goal['name'])
            for action in goal['actions']:
                success = robot.run(action)
                self.app['logger'].info("Done") if success else self.app['logger'].info("Failed")
                
